src/mongocxx/options/coll_mod.hpp

```diff
@@ -0,0 +1,83 @@
+// Copyright 2015 MongoDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <mongocxx/config/prelude.hpp>
+
+#include <bsoncxx/document/value.hpp>
+#include <bsoncxx/stdx/optional.hpp>
+#include <mongocxx/stdx.hpp>
+#include <mongocxx/validation_criteria.hpp>
+
+namespace mongocxx {
+MONGOCXX_INLINE_NAMESPACE_BEGIN
+namespace options {
+
+///
+/// Class representing the optional arguments to a MongoDB createMod command.
+///
+class MONGOCXX_API coll_mod {
+   public:
+    ///
+    /// The index flag changes the expiration time of a TTL collection.
+    ///
+    /// @param index_spec the spec for an existing index in the collection
+    /// @param seconds number of seconds to subtract from the current time
+    ///
+    /// @see https://docs.mongodb.org/manual/tutorial/expire-data/
+    ///
+    void index(bsoncxx::document::view_or_value index_spec, int seconds);
+
+    ///
+    /// When true, disables the power of 2 sizes allocation for the collection.
+    ///
+    /// @see: https://docs.mongodb.org/manual/reference/method/db.createCollection/
+    ///
+    /// @param no_padding
+    ///   When true, disables power of 2 sizing for this collection.
+    ///
+    void no_padding(bool no_padding);
+
+    ///
+    /// Specify validation criteria for this collection.
+    ///
+    /// TODO: link to documentation once it is published.
+    ///
+    /// @param validation
+    ///   Validation criteria for this collection.
+    ///
+    void validation_criteria(class validation_criteria validation);
+
+    ///
+    /// Return a bson document representing a collMod command with the given options
+    /// set on this object.
+    ///
+    /// @return Options, as a document.
+    ///
+    bsoncxx::document::value to_document() const;
```

Comment:
This is fine for now. I have some thoughts about how I want object to document and document to object conversion to work, overall. We can tackle that as part of https://jira.mongodb.org/browse/CXX-763, which I now think we will probably want to do.